### ISSUE TITLE
Waiting for peers. Connected peers 0

### ISSUE THREAD

**Issue by rudijs** [mimblewimble/grin#2760]
**Describe the bug**
All default grin node waiting for peers and connected peers is 0.

I've tried the current 1.0.3, 1.0.2 and the current beta but my node just sits with zero connections and no syncing.

I would just like to run a validating node, no mining or anything.

My goal is to receive, store and send some Grin coin.

I haven't yet used the wallet, I understand I need a local node first.

**To Reproduce**
1. I'm following the readme instructions, just unzip and run ./grin (no compile)

2. Expecting to see peers connect and downloading of the mainnet chain.

3. I don't see any errors with any of the versions I've tried

**Relevant Information**

```
~/.g/main ❯❯❯ du -sh *
152K    chain_data
4.0K    grin-server.log
8.0K    grin-server.toml

~/.g/main ❯❯❯ ll chain_data
total 16K
drwxr-xr-x 4 rudi rudi 4.0K Apr 18 09:44 header
drwxr-xr-x 2 rudi rudi 4.0K Apr 18 09:44 lmdb
drwxr-xr-x 2 rudi rudi 4.0K Apr 18 09:44 peer
drwxr-xr-x 5 rudi rudi 4.0K Apr 18 09:44 txhashset
-rw-r--r-- 1 rudi rudi    0 Apr 18 09:44 grin.lock

~/.g/main ❯❯❯ cat grin-server.log
20190418 09:44:17.149 INFO grin_util::logger - log4rs is initialized, file level: Info, stdout level: Warn, min. level: Info
20190418 09:44:17.149 INFO grin - Using configuration file at /home/rudi/.grin/main/grin-server.toml
20190418 09:44:17.149 INFO grin - This is Grin version 1.0.3 (git v1.0.3), built for x86_64-unknown-linux-gnu by rustc 1.33.0 (2aa4c46cf 2019-02-28).
20190418 09:44:17.149 WARN grin::cmd::server - Starting GRIN in UI mode...
20190418 09:44:17.149 INFO grin_servers::grin::server - Starting server, genesis block: 40adad0aec27
20190418 09:44:17.218 INFO grin_chain::chain - init: saved genesis: 40adad0aec27
20190418 09:44:17.246 INFO grin_servers::grin::server - Starting rest apis at: 127.0.0.1:3413
20190418 09:44:17.246 INFO grin_api::handlers - Starting HTTP API server at 127.0.0.1:3413.
20190418 09:44:17.246 INFO grin_servers::grin::server - Starting dandelion monitor: 127.0.0.1:3413
20190418 09:44:17.246 WARN grin_servers::grin::server - Grin server started.
```

**Screenshots**
![grin](https://user-images.githubusercontent.com/1711940/56333736-8afd1380-61bf-11e9-9f7f-7cb83034e016.png)

**Desktop (please complete the following information):**
OS: Standard Ubuntu Desktop

```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.2 LTS"
```
**Additional context**
I also tried without joy the suggested from another issue:

`TMPDIR=/home/rudi/Downloads/grin/tmp ./grin`

I've also tried with and without a VPN, both without success.

**Comment by rudijs** [comment]
Ah, when I set `file_log_level = "Debug"` I get some errors:


```
20190418 10:47:34.227 INFO grin_util::logger - log4rs is initialized, file level: Debug, stdout level: Debug, min. level: Debug
20190418 10:47:34.227 INFO grin - Using configuration file at /home/rudi/.grin/main/grin-server.toml
20190418 10:47:34.227 INFO grin - This is Grin version 1.0.3 (git v1.0.3), built for x86_64-unknown-linux-gnu by rustc 1.33.0 (2aa4c46cf 2019-02-28).
20190418 10:47:34.227 DEBUG grin - Built with profile "release", features "".
20190418 10:47:34.227 WARN grin::cmd::server - Starting GRIN in UI mode...
20190418 10:47:34.227 INFO grin_servers::grin::server - Starting server, genesis block: 40adad0aec27
20190418 10:47:34.229 DEBUG grin_chain::txhashset::txhashset - Rewind to header 40adad0aec27 at 0
20190418 10:47:34.229 DEBUG grin_chain::txhashset::txhashset - txhashset: rewind_to_pos: header 1, output 1, kernel 1
20190418 10:47:34.229 DEBUG grin_chain::chain - init: rewinding and validating before we start... 40adad0aec27 at 0
20190418 10:47:34.256 DEBUG grin_chain::chain - init: head: 17179869184 @ 0 [40adad0aec27]
20190418 10:47:34.256 DEBUG grin_chain::chain - init: header_head: 17179869184 @ 0 [40adad0aec27]
20190418 10:47:34.256 DEBUG grin_chain::chain - init: sync_head: 17179869184 @ 0 [40adad0aec27]
20190418 10:47:34.256 DEBUG grin_chain::txhashset::txhashset - txhashset: starting compaction...
20190418 10:47:34.256 DEBUG grin_chain::txhashset::txhashset - txhashset: check_compact output mmr backend...
20190418 10:47:34.257 DEBUG grin_chain::txhashset::txhashset - txhashset: check_compact rangeproof mmr backend...
20190418 10:47:34.257 DEBUG grin_chain::txhashset::txhashset - txhashset: ... compaction finished
20190418 10:47:34.257 DEBUG grin_chain::txhashset::txhashset - txhashset: rebuild_index: 1 UTXOs, took 0s
20190418 10:47:34.278 DEBUG grin_chain::chain - remove_historical_blocks: head height: 0, tail height: 0, horizon: 10080, cutoff: 0
20190418 10:47:34.283 DEBUG grin_servers::common::types - sync_state: sync_status: Initial -> AwaitingPeers(true)
20190418 10:47:34.284 INFO grin_servers::grin::server - Starting rest apis at: 127.0.0.1:3413
20190418 10:47:34.284 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin-tech.org
20190418 10:47:34.284 INFO grin_api::handlers - Starting HTTP API server at 127.0.0.1:3413.
20190418 10:47:34.284 INFO grin_servers::grin::server - Starting dandelion monitor: 127.0.0.1:3413
20190418 10:47:34.284 DEBUG grin_servers::grin::dandelion_monitor - Started Dandelion transaction monitor.
20190418 10:47:34.284 WARN grin_servers::grin::server - Grin server started.
20190418 10:47:34.529 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.icu
20190418 10:47:34.738 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.713.mw
20190418 10:47:35.229 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.lesceller.com
20190418 10:47:35.564 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns mainnet.seed.grin.prokapi.com
20190418 10:47:35.687 DEBUG grin_servers::grin::seed - Retrieving seed nodes from dns grinseed.yeastplume.org
20190418 10:47:35.741 DEBUG grin_servers::grin::seed - Retrieved seed addresses: [PeerAddr(V4(95.216.163.175:3414)), PeerAddr(V4(145.239.4.27:3414)), PeerAddr(V4(46.4.91.48:3414)), PeerAddr(V4(35.247.33.125:3414)), PeerAddr(V4(5.9.152.75:3414)), PeerAddr(V4(198.245.50.26:3414)), PeerAddr(V4(167.99.3.23:3414)), PeerAddr(V4(139.162.168.18:3414)), PeerAddr(V4(45.118.135.254:3414)), PeerAddr(V4(35.226.50.239:3414)), PeerAddr(V4(173.230.154.243:3414)), PeerAddr(V4(37.187.124.202:3414)), PeerAddr(V4(159.65.11.32:3414)), PeerAddr(V4(157.230.142.11:3414)), PeerAddr(V4(94.130.229.193:3414)), PeerAddr(V4(35.181.35.93:3414)), PeerAddr(V4(157.230.48.217:3414)), PeerAddr(V4(18.204.166.78:3414)), PeerAddr(V4(34.207.234.131:3414)), PeerAddr(V4(195.201.38.140:3414)), PeerAddr(V4(109.74.202.16:3414))]
20190418 10:47:35.742 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20190418 10:47:35.742 DEBUG grin_servers::grin::seed - monitor_peers: no dandelion relay updating
20190418 10:47:35.742 DEBUG grin_p2p::peers - Could not update dandelion relay
20190418 10:47:36.019 DEBUG grin_p2p::peer - connect: handshaking with Ok(V4(35.247.33.125:3414))
20190418 10:47:36.019 DEBUG grin_p2p::peer - connect: handshaking with Ok(V4(35.181.35.93:3414))
20190418 10:47:36.019 DEBUG grin_p2p::peer - connect: handshaking with Ok(V4(18.204.166.78:3414))
20190418 10:47:36.211 DEBUG grin_p2p::peer - connect: handshaking with Err(Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" }) failed with error: Connection(Custom { kind: ConnectionAborted, error: StringError("read_exact") })
20190418 10:47:36.211 DEBUG grin_p2p::peer - Error shutting down conn: Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" }
20190418 10:47:37.743 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20190418 10:47:37.743 DEBUG grin_servers::grin::seed - monitor_peers: no dandelion relay updating
20190418 10:47:37.743 DEBUG grin_p2p::peers - Could not update dandelion relay
20190418 10:47:41.743 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20190418 10:47:41.743 DEBUG grin_servers::grin::seed - monitor_peers: no dandelion relay updating
20190418 10:47:41.743 DEBUG grin_p2p::peers - Could not update dandelion relay
20190418 10:47:49.744 DEBUG grin_servers::grin::seed - monitor_peers: on 0.0.0.0:3414, 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
20190418 10:47:49.744 DEBUG grin_servers::grin::seed - monitor_peers: no dandelion relay updating
20190418 10:47:49.744 DEBUG grin_p2p::peers - Could not update dandelion relay
```

**Comment by antiochp** [comment]
```
20190418 10:47:35.741 DEBUG grin_servers::grin::seed - Retrieved seed addresses: [PeerAddr(V4(95.216.163.175:3414)), PeerAddr(V4(145.239.4.27:3414)), PeerAddr(V4(46.4.91.48:3414)), PeerAddr(V4(35.247.33.125:3414)), PeerAddr(V4(5.9.152.75:3414)), PeerAddr(V4(198.245.50.26:3414)), PeerAddr(V4(167.99.3.23:3414)), PeerAddr(V4(139.162.168.18:3414)), PeerAddr(V4(45.118.135.254:3414)), PeerAddr(V4(35.226.50.239:3414)), PeerAddr(V4(173.230.154.243:3414)), PeerAddr(V4(37.187.124.202:3414)), PeerAddr(V4(159.65.11.32:3414)), PeerAddr(V4(157.230.142.11:3414)), PeerAddr(V4(94.130.229.193:3414)), PeerAddr(V4(35.181.35.93:3414)), PeerAddr(V4(157.230.48.217:3414)), PeerAddr(V4(18.204.166.78:3414)), PeerAddr(V4(34.207.234.131:3414)), PeerAddr(V4(195.201.38.140:3414)), PeerAddr(V4(109.74.202.16:3414))]
```

You're definitely receiving _some_ p2p messages via the network.

Can you try removing the `chain_data/peer` and restarting your node?

**Comment by rudijs** [comment]
Hi @antiochp 

Yep that did it - thanks!

I simply `rm -rf peer/*` (files only not folder) then restarted and boom peers connected.

Quickly to over 100+ connected peers and syncing fast.

My plan is, as a user, to send and receive a few times, then try a delete and recover from seed phrase.

Then fully delete and re-download the blockchain, then check balance.

Basically a technical savvy end-user in newbie mode for this cryptocurrency.

Also I do have a firewall up (ufw on ubuntu) but I am allowing inbound 3414.

Thanks again.

**Comment by antiochp** [comment]
We had some issues with peers filling up all their connections and not accepting any new connections.
And I suspect you got unlucky - your peer list on startup filled up with peers that would actually accept connections.
Clearing the `peer` dir forced your node to go and ask for a full set of new peers.

Glad it worked!
